GameServerLib/API/ApiFunctionManager.cs

```diff
@@ -659,6 +664,20 @@ public static void SpellCast(IObjAiBase caster, int slot, SpellSlotType slotType
             SpellCast(caster, slot, slotType, target.Position, target.Position, fireWithoutCasting, overrideCastPos, new List<ICastTarget> { castTarget }, isForceCastingOrChanneling, overrideForceLevel, updateAutoAttackTimer, useAutoAttackSpell);
         }
 
+        public static void SpellCastItem(IObjAiBase caster, string itemSpellName, bool fireWithoutCasting, IAttackableUnit target, Vector2 overrideCastPos, bool isForceCastingOrChanneling = false, int overrideForceLevel = -1, bool updateAutoAttackTimer = false, bool useAutoAttackSpell = false)
+        {
+            // Apply the spell to the TempItemSlot.
+            caster.SetSpell(itemSpellName, (byte)SpellSlotType.TempItemSlot, true);
+            SpellCast(caster, 0, SpellSlotType.TempItemSlot, fireWithoutCasting, target, overrideCastPos, isForceCastingOrChanneling, overrideForceLevel, updateAutoAttackTimer, useAutoAttackSpell);
+        }
+
+        public static void SpellCastItem(IObjAiBase caster, string itemSpellName, Vector2 pos, Vector2 endPos, bool fireWithoutCasting, Vector2 overrideCastPos, bool isForceCastingOrChanneling = false, int overrideForceLevel = -1, bool updateAutoAttackTimer = false, bool useAutoAttackSpell = false)
+        {
+            // Apply the spell to the TempItemSlot.
+            caster.SetSpell(itemSpellName, (byte)SpellSlotType.TempItemSlot, true);
+            SpellCast(caster, 0, SpellSlotType.TempItemSlot, pos, endPos, fireWithoutCasting, overrideCastPos, null, isForceCastingOrChanneling, overrideForceLevel, updateAutoAttackTimer, useAutoAttackSpell);
```

Comment:
What is the 'null' here for?

Reply:
castTarget variable.

Reply:
But it doesn't appear in the lines before. (See the function above this one)
It is like all the parameters are shifted in ones of the calls to SpellCast.

Reply:
Seems like there are two very similar SpellCast functions signatures. I think we should pass all the parameters in sort of container and not like that as the code tends to be very confusing in this current way (10+ parameters).

Reply:
The above function uses the other SpellCast which is target only, so it doesn't have position variables.
This function uses the original SpellCast function which has both position and target variables.